src/Symfony/Component/Profiler/DataCollector/DataCollectorInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Profiler\DataCollector;
+
+/**
+ * DataCollectorInterface.
+ *
+ * @author Jelte Steijaert <jelte@khepri.be>
+ */
+interface DataCollectorInterface
+{
+    /**
+     * Set the Token of the active profile.
+     *
+     * @param $token
+     */
+    public function setToken($token);
+
+    /**
+     * Returns the name of the collector.
+     *
+     * @return string The collector name
+     *
+     * @api
+     */
+    public function getName();
```

Comment:
Why do Data collectors need names? What's the use case for accessing collectors by name? I think that this method should be moved to ProfileDataInterface.